### PR TITLE
Fixing some bugs for the new deployment UI

### PR DESCRIPTION
**Pull request by stojg**

### CODE REVIEW

File: code/control/DeployDispatcher.php [hunk 1]
```diff
@@ -124,9 +124,15 @@ public function show(SS_HTTPRequest $request) {
 	 * @return \SS_HTTPResponse
 	 */
 	public function log(SS_HTTPRequest $request) {
-		$deployment = DNDeployment::get()->byId($request->params('ID'));
-		if (!$deployment || !$deployment->exists) {
-			return $this->getAPIResponse(['message' => 'This deployment does not exist'], 404);
+		$deployID = $request->param('ID');
+		$deployment = DNDeployment::get()->byId($deployID);
```
Comment (halkyon) [inline code/control/DeployDispatcher.php]: Can you copy the way show action does this so it's consistent? just copy/paste it including the 404 response

File: code/control/DeployDispatcher.php [hunk 2]
```diff
@@ -256,5 +251,23 @@ protected function getStackMemberData(Member $member) {
...
+	
+	/**
+	 * Check if a DNDeployment is exists and do permission checks on it. If there is something wrong it will return
+	 * an APIResponse with the error, otherwise null.
+	 *
+	 * @param DNDeployment $deployment
+	 *
+	 * @return null|SS_HTTPResponse
+	 */
+	protected function validateDeployement(\DNDeployment $deployment) {
```
Comment (halkyon) [inline code/control/DeployDispatcher.php]: misspelled deployment

File: code/control/DeployDispatcher.php [hunk 3]
```diff
@@ -256,5 +251,23 @@ protected function getStackMemberData(Member $member) {
 			'name' => $member->getName()
 		];
 	}
+	
+	/**
+	 * Check if a DNDeployment is exists and do permission checks on it. If there is something wrong it will return
+	 * an APIResponse with the error, otherwise null.
+	 *
+	 * @param DNDeployment $deployment
```
Comment (halkyon) [inline code/control/DeployDispatcher.php]: should be `@param \DNDeployment $deployment` (missing the \)

File: code/control/DeployDispatcher.php [hunk 4]
```diff
@@ -256,5 +251,23 @@ protected function getStackMemberData(Member $member) {
 			'name' => $member->getName()
 		];
 	}
+	
+	/**
+	 * Check if a DNDeployment is exists and do permission checks on it. If there is something wrong it will return
```
Comment (halkyon) [inline code/control/DeployDispatcher.php]: "Check if a DNDeployment exists" (drop the a)

File: code/control/DeployDispatcher.php [hunk 5]
```diff
@@ -36,6 +29,13 @@ class DeployDispatcher extends Dispatcher {
 	 */
 	protected $environment = null;
 
+	/**
```
Comment (stojg) [inline code/control/DeployDispatcher.php]: Moved because it has private visibility

File: code/control/DeployDispatcher.php [hunk 6]
```diff
@@ -257,4 +252,22 @@ protected function getStackMemberData(Member $member) {
...
+	 * Check if a DNDeployment exists and do permission checks on it. If there is something wrong it will return
+	 * an APIResponse with the error, otherwise null.
+	 *
+	 * @param \DNDeployment $deployment
+	 *
+	 * @return null|SS_HTTPResponse
+	 */
+	protected function validateDeployments(\DNDeployment $deployment) {
```
Comment (halkyon) [inline code/control/DeployDispatcher.php]: it's just singular ;) so validateDeployment

**Pull request merged**